doc/en/user/source/extensions/css/workshop/polygon.rst

```diff
@@ -0,0 +1,880 @@
+
+For our Polygon exercises we will try and limit our CSS documents to a single rule, in order to showcase the properties used for rendering.
+
+Reference:
+
+* `Polygon Symbology <../properties.html#polygon-symbology>`_ (User Manual | CSS Property Listing)
+* `Polygons <../cookbook/polygon.html>`_ (User Manual | CSS Cookbook)
+* `Polygon <../../../styling/sld-reference/polygonsymbolizer.html>`_ (User Manual | SLD Reference )
+
+This exercise makes use of the ``ne:states_provinces_shp`` layer.
+
+#. Navigate to :menuselection:`CSS Styles`.
+
+#. Set ``ne:states_provinces_shp`` as the preview layer.
+
+   .. image:: img/polygon_01_preview.png
+
+#. Create a new CSS style :kbd:`polygon_example`.
+
+   .. list-table:: 
+      :widths: 30 70
+      :stub-columns: 1
+
+      * - Workspace for new layer:
+        - :kbd:`No workspace`
+      * - New style name:
+        - :kbd:`polygon_example`
+     
+   .. image:: img/polygon_02_create.png
+
+#. An initial style is provided:
+
+   .. code-block:: css
+   
+      * { fill: lightgrey; }
+
+#. Click on the tab :guilabel`Map` to preview.
+
+   .. image:: img/polygon_04_preview.png
+
+Stroke and Fill
+---------------
+
+The **key property** for polygon data is **fill**.
+
+.. image:: img/PolygonFill.svg
+
+The **fill** property is used to provide the color, or pattern, used to draw the interior of a polygon.
+
+
+#. Replace the contents of ``polygon_example`` with the following **fill** example:
+
+   .. code-block:: css
+   
+      * {
+         fill: gray;
+      }
+
+#. The :guilabel:`Map` tab can be used preview the change:
+
+   .. image:: img/polygon_fill_1.png
+
+#. To draw the boundary of the polygon the **stroke** property is used:
+
+   The **stroke** property is used to provide the color, or pattern, for the polygon boundary. It is effected by the same parameters (and vendor specific parameters) as used for LineStrings. 
+   
+   .. code-block:: css
+      :emphasize-lines: 3,4
+      
+      * {
+         fill: gray;
+         stroke: black;
+         stroke-width: 2;
+      }
+   
+   .. note:: Technically the boundary of a polygon is a specific case of a LineString where the first and last vertex are the same, forming a closed LinearRing.
+
+#. The effect of adding **stroke** is shown in the map preview:
+   
+   .. image:: img/polygon_fill_2.png
+
+#. An interesting technique when styling polygons in conjunction with background information is to control the fill opacity.
+
+   The **fill-opacity** property is used to adjust transparency (provided as range from 0.0 to 1.0). Use of **fill-opacity** to render polygons works well in conjunction with a raster base map. This approach allows details of the base map to shown through.
+
+   The **stroke-opacity** property is used in a similar fashion, as a range from 0.0 to 1.0.
+
+   .. code-block:: css
+      :emphasize-lines: 3,6
+      
+      * {
+         fill: white;
+         fill-opacity: 50%;
```

Comment:
If ``fill-opacity`` is a decimal value from 0.0 to 1.0, is this incorrect, or do we need to explain that percentages are also allowed?